### ISSUE TITLE
Inquiry about "CacheableDataset from wrapper"

### ISSUE THREAD

**Issue by AhmedHussKhalifa** [yoshitomo-matsubara/torchdistill#165]
Hey,

I am trying to tailor your framework to accommodate my experiments on imagenet. One of my experiments is to implement this paper [Knowledge distillation: A good teacher is patient and consistent](https://arxiv.org/abs/2106.05237) where they do some data augmentation at the teacher and student side. Each model has different input data. I expect I  would need to creat each of them a different dataloader. I will also maintain the same augmentation method for each image to be able to store the output vector of the teacher using [default_idx2subpath](https://github.com/yoshitomo-matsubara/torchdistill/blob/f7e757f63982c76bc3e9db4354edf1e008963f8a/torchdistill/datasets/wrapper.py#L14) function on my hard drive (SSD). I figured that saving the data on the SSD compared to loading the images made the training runs slower. 

I am trying to train a Student model -ResNet18- using 2 GPUs on ImageNet and a teacher model -Resnet34-. Could u recommend what is the best scenario for this pipeline to run the code faster giving the current resources?

Another question, if I need to train using cosine scheduler [CosineAnnealingLR ] with Adam optimizer, which method should I change?

**Comment by yoshitomo-matsubara** [comment]
@AhmedHussKhalifa Closing this issue as I haven't seen any follow-up for a while.
Open a new `Discussion` (not `Issue`) if you still have questions

**Comment by AhmedHussKhalifa** [comment]
Thank you @yoshitomo-matsubara.
